### PR TITLE
Improve S2 chromatic story coverage

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@react-spectrum/s2/chromatic/ActionButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+};
+
+// TODO: eventually make all none S2 chromatic stories have
+// chromatic: {
+//   disableSnapshot: true
+// }
+// so that we centralize the chromatic stories. Wait until all these chromatic stories are finished
+// and we run chromatic one last time to ensure a good baseline, then have all the stories in the
+// chromatic folder import the original stories and rexport them
```
Comment (LFDanLu) [inline packages/@react-spectrum/s2/chromatic/ActionButton.stories.tsx:112]: As we add powerset stories, some/all of the currently snapshotted S2 stories are redundant/extraneous so we'll want to centralize them all into this folder eventually

File: packages/@react-spectrum/s2/chromatic/ActionButton.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    chromaticProvider: {disableAnimations: true}
+  },
+  title: 'S2 Chromatic/ActionButton'
+};
+
+export default meta;
+
+let states = [
+  {isQuiet: true},
+  {isReadOnly: true},
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/ActionButton.stories.tsx:34]: whoops, thanks copy pasta

File: packages/@react-spectrum/s2/chromatic/ActionMenu.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+export default meta;
+type Story = StoryObj<typeof ActionMenu<any>>;
+
+export const Static: Story = {...Example};
+
+Static.play = async ({canvasElement}) => {
+  await userEvent.tab();
+  await userEvent.keyboard('[Enter]');
+  let body = canvasElement.ownerDocument.body;
+  await within(body).findByRole('menu');
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/ActionMenu.stories.tsx:35]: I'm not sure if I can since these interactions come from `'@storybook/testing-library';`, will have to see if storybook is doing anything special on top of the userEvent api. I'll handle that in a future PR if turns out I can use the test utils, don't want this PR held up by that one.

The reason for tab and enter is to get the focus ring, figured it was better to test for that style than just doing the click

File: packages/@react-spectrum/s2/chromatic/Avatar.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  title: 'S2 Chromatic/Avatar'
+};
+
+export default meta;
+
+export const Default = {
+  render: () => <Avatar alt="design provided" src="https://i.imgur.com/xIe7Wlb.png" />
+};
+
+export const OverBackground = {
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/Avatar.stories.tsx:31]: not quite sure the ask here, the base story goes through all the layer backgrounds if that is what is being asked for here
image

File: packages/@react-spectrum/s2/chromatic/BreadcrumbsRTL.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+const meta: Meta<typeof Breadcrumbs> = {
+  component: Breadcrumbs,
+  parameters: {
+    chromaticProvider: {colorSchemes: ['dark'], backgrounds: ['base'], locales: ['ar-AE'], disableAnimations: true}
+  },
+  title: 'S2 Chromatic/BreadcrumbsRTL'
+};
+
+export default meta;
+export {Dynamic, Powerset} from './Breadcrumbs.stories';
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/BreadcrumbsRTL.stories.tsx:26]: Ah this is because I've set
```
Dynamic.parameters = {
  // TODO: add ar-AE when we get intl strings
  chromaticProvider: {colorSchemes: ['light'], backgrounds: ['base'], locales: ['en-US'], disableAnimations: true}
};
```
In the Breadcrumb stories that they use to make sure `ar-AE` doesn't actually apply since it blows up the story due to missing intl strings. When those strings are added I'll update these, will add a comment

File: packages/@react-spectrum/s2/chromatic/Button.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  let {children, ...otherArgs} = args;
+  return (
+    <div className={style({display: 'grid', gridTemplateColumns: 'repeat(4, minmax(0, 250px))', gridAutoFlow: 'row', alignItems: 'center', justifyItems: 'start', gap: 24, width: '[100vw]'})}>
+      {combinations.map(c => {
+        let key = Object.keys(c).map(k => shortName(k, c[k])).join(' ');
+        if (!key) {
+          key = 'default';
+        }
+
+        let button = <Button key={key} {...otherArgs} {...c}>{children ? children : key}</Button>;
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/Button.stories.tsx:52]: sure I can do that, I originally just wanted to display them visibly so someone going through chromatic changes didn't have to open the build separately to check the config

File: packages/@react-spectrum/s2/chromatic/ButtonGroup.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+          </ButtonGroup>
+        );
+      })}
+    </div>
+  );
+};
+
+let horizontalStates = [
+  {isDisabled: true},
+  {size: ['S', 'M', 'L', 'XL']},
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/ButtonGroup.stories.tsx:58]: I guess but it would be good to know if if a size is no longer available or for some reason stopped working?

File: packages/@react-spectrum/s2/chromatic/Checkbox.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+export const DefaultPt3 = {
+  render: Template,
+  args: {
+    combos: chunks[2]
+  }
+};
+
+export {LongLabel};
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/Checkbox.stories.tsx:77]: sure I can add it to the list that @yihuiliao and I are tracking locally

File: packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+export default meta;
+
+let states = [
+  {isDisabled: true},
+  {isEmphasized: true},
+  {isInvalid: true},
+  {isReadOnly: true},
+  {isRequired: true},
+  // {labelAlign: ['start', 'end']},
+  // {labelPosition: ['top', 'side']},
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx:35]: commented this out for now since it proved difficult to implement with the powerset given the sizing and/or label size. Probably should have a subset of dedicated stories but to be done later

File: packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+};
+
+export default meta;
+
+let states = [
+  {isDisabled: true},
+  {isEmphasized: true},
+  {isInvalid: true},
+  {isReadOnly: true},
+  {isRequired: true},
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx:33]: my only hesitation is that this assumes that the group wrapper styling won't ever be affected by these props. Happy for others to weigh in, I know the power sets are already quite hard to parse so happy to be overruled here

File: packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+  // {labelAlign: ['start', 'end']},
+  // {labelPosition: ['top', 'side']},
+  {necessityIndicator: ['label', 'icon']},
+  {size: ['S', 'M', 'L', 'XL']}
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/CheckboxGroup.stories.tsx:37]: again, just a bit wary that this assumes that the set of sizes will always be in lockstep between the Checkbox and CheckboxGroup thus we can skip rendering all of them.

File: packages/@react-spectrum/s2/chromatic/ColorArea.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  parameters: {
+    layout: 'centered'
+  },
+  tags: ['autodocs'],
+  title: 'S2 Chromatic/ColorArea'
+};
+
+export default meta;
+
+export const Default = {
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/ColorArea.stories.tsx:27]: yep, on our list for stories to add

File: packages/@react-spectrum/s2/chromatic/ColorField.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+};
+
+export const LabelPositionTopPt5 = {
+  render: Template,
+  args: {
+    ...LabelPositionTop.args,
+    combos: topChunks[4]
+  }
+};
+
```
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-spectrum/s2/chromatic/ColorField.stories.tsx:125]: good catch! I'll make a ticket documenting these

File: packages/@react-spectrum/s2/chromatic/Button.stories.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/s2/chromatic/Button.stories.tsx]: Yeah, that's nice when possible as well, but if there isn't room, I'm fine for them to be data-attributes. I think we have a blend in the RSP existing chromatic as well?

File: packages/@react-spectrum/s2/chromatic/ActionMenu.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/s2/chromatic/ActionMenu.stories.tsx:35]: ahhh, makes sense, thanks for explaining

File: packages/@react-spectrum/s2/chromatic/Avatar.stories.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/s2/chromatic/Avatar.stories.tsx:31]: Avatars can have a border around them that matches the layer color
```
      base
      'layer-1'
      'layer-2'
      pasteboard
      elevated
```
But I don't think they have that border unless in a Card or AvatarGroup?

File: packages/@react-spectrum/s2/chromatic/ButtonGroup.stories.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/s2/chromatic/ButtonGroup.stories.tsx:59]: yeah, but we don't need them in the powerset then
anything to get smaller sets :)

also, we'd know through the Typescript and through the chromatic tests for buttons which include all the sizes

**Review by snowystinger — APPROVED** [review]
happy for anything else to be followup, would be good to get testing in place asap
